Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver/handlers_test.go

```diff
@@ -41,13 +42,21 @@ func (f fakeRL) CanAccept() bool { return bool(f) }
 func (f fakeRL) Accept()         {}
 
 func expectHTTP(url string, code int, t *testing.T) {
-	r, err := http.Get(url)
+	var response *http.Response
+	err := wait.Poll(10, time.Duration(100*time.Millisecond), func() (bool, error) {
+		var e error
+		response, e = http.Get(url)
+		if response.StatusCode == 429 {
+			return false, e
```

Comment:
If is non-nil (likely), this will cause your test to fail.  If you want to reenter, you can `return false, nil`.  If the timeout is reached, `Poll` produces an error, so you'll be covered.